db/view/view.cc

```diff
@@ -438,18 +438,22 @@ class data_query_result_builder {
     }
 };
 
-bool matches_view_filter(const schema& base, const view_info& view, const partition_key& key, const clustering_row& update, gc_clock::time_point now) {
+bool matches_view_filter(const schema& base, const view_info& view, const partition_key& key, const clustering_or_static_row& update, gc_clock::time_point now) {
     auto slice = make_partition_slice(base);
 
     data_query_result_builder builder(base, slice);
     builder.consume_new_partition(dht::decorate_key(base, key));
-    builder.consume(clustering_row(base, update), row_tombstone{}, update.is_live(base, tombstone{}, now));
+    if (update.is_clustering_row()) {
+        builder.consume(clustering_row(base, update.as_clustering_row(base)), row_tombstone{}, update.is_live(base, tombstone(), now));
+    } else {
+        builder.consume(static_row(base, update.as_static_row(base)), tombstone{}, update.is_live(base, tombstone(), now));
+    }
     builder.consume_end_of_partition();
     auto result = builder.consume_end_of_stream();
     view_filter_checking_visitor visitor(base, view);
     query::result_view::consume(result, slice, visitor);
 
-    return clustering_prefix_matches(base, view, key, update.key())
+    return (update.is_static_row() || clustering_prefix_matches(base, view, key, *update.key()))
```

Comment:
This code is usually reached quite frequently, see `create_entry`, `delete_old_entry`, `update_entry` - they all call `matches_view_filter`. However, the current implementation of `view_filter_checking_visitor` seems to just do nothing when a static row is processed: the `result` it visits doesn't contain static rows, so `accept_new_row` which calculates the condition is not called.

As you mentioned, there isn't a scenario right now where evaluating the condition would be necessary, so instead of future-proofing it now (which I'm not sure how to do correctly) I've put a TODO here and changed this function to always `return true` for static row updates.